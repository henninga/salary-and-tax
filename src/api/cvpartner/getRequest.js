import { config } from 'dotenv';
import rp from 'request-promise';

config();

export const BASE_URL = `https://technologic.cvpartner.com`;

const getRequest = async url => await rp({
    url: BASE_URL + url, json: true, headers: { 'Authorization': `Token token="${process.env.CV_PARTNER_API_KEY}"` },
});

export default getRequest;
