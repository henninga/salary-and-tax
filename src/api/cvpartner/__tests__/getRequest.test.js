import rp from 'request-promise';
import getRequest, { BASE_URL } from '../getRequest';

const mockData = [{ name: 'CV Partner', id: 'userId', default_cv_id: 'cvId' }];

jest.mock('request-promise');
rp.mockImplementation(() => Promise.resolve(mockData));

describe('getRequest', () => {
    it('should resolve to matching snapshot', async () => {
        await expect(getRequest('/api/call')).resolves.toMatchSnapshot();
    });
    it('should reject to matching snapshot', async () => {
        rp.mockImplementationOnce(() => Promise.reject({ error: 'error message' }));
        await expect(getRequest('/api/call')).rejects.toMatchSnapshot();
    });
    it('should add correct url', async () => {
        const apiUrl = '/api/url/to/be/appended';
        await getRequest(apiUrl);
        expect(rp).toHaveBeenCalledWith(expect.objectContaining({ url: expect.stringContaining(apiUrl) }));
        expect(rp).toHaveBeenCalledWith(expect.objectContaining({ url: expect.stringContaining(BASE_URL) }));
    });
    it('should add headers', async () => {
        await getRequest('/api/url/to/be/appended');
        expect(rp).toHaveBeenCalledWith(expect.objectContaining({
            headers: expect.objectContaining({ Authorization: expect.stringContaining('Token token=') }),
        }));
    });
});