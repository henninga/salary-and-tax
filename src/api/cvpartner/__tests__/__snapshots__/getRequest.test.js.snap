// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getRequest should reject to matching snapshot 1`] = `
Object {
  "error": "error message",
}
`;

exports[`getRequest should resolve to matching snapshot 1`] = `
Array [
  Object {
    "default_cv_id": "cvId",
    "id": "userId",
    "name": "CV Partner",
  },
]
`;
